apollo-router/src/services/connector_service.rs

```diff
@@ -73,24 +74,32 @@ async fn execute(
     schema: &Valid<apollo_compiler::Schema>,
 ) -> Result<ConnectResponse, BoxError> {
     let context = request.context.clone();
+    let context2 = context.clone();
     let original_subgraph_name = connector.id.subgraph_name.to_string();
 
+    let mut debug = context
+        .extensions()
+        .with_lock(|mut lock| lock.remove::<ConnectorContext>());
+
     let requests =
-        make_requests(request, connector, schema).map_err(|_e| BoxError::from("TODO"))?;
+        make_requests(request, connector, &mut debug).map_err(|_e| BoxError::from("TODO"))?;
 
     let tasks = requests.into_iter().map(move |(req, key)| {
         let context = context.clone();
         let original_subgraph_name = original_subgraph_name.clone();
         async move {
+            let context = context.clone();
+
             let client = http_client_factory.create(&original_subgraph_name);
             let req = HttpRequest {
                 http_request: req,
-                context,
+                context: context.clone(),
```

Comment:
Why the double clone here?

```suggestion
                context,
```

Reply:
leftover from refactoring, good catch!